src/orion/client/experiment.py

```diff
@@ -105,16 +90,25 @@ def __init__(self, experiment, producer, executor=None, heartbeat=None):
         if heartbeat is None:
             heartbeat = orion.core.config.worker.heartbeat
         self.heartbeat = heartbeat
-        self.executor = executor or executor_factory.create(
-            orion.core.config.worker.executor,
-            n_workers=orion.core.config.worker.n_workers,
-            **orion.core.config.worker.executor_configuration,
-        )
+        self._executor = executor
+        self._executor_owner = False
         self.plot = PlotAccessor(self)
 
     ###
     # Attributes
     ###
+    @property
```

Comment:
I like this! Is it tested? I can't find one.

Reply:
Looks like tests are still missing for this.

Reply:
added

Reply:
Where? I did not find it. :(

Reply:
https://github.com/Epistimio/orion/pull/684/files#diff-f06b94773052cdf9a38bfdb3ea74cb852af76b54d9c1acf6841cd98db127d04cR618

Reply:
https://github.com/Epistimio/orion/blob/4b25e064b7c3725783a1a62f656a7dd443381a63/tests/unittests/client/test_runner.py#L640

Reply:
Great, thanks! I think this test would below in test_experiment_client however since it is not testing anything about the runner. It's different for all workon tests, which are involving the runner.

Reply:
moved